Framework/Nexus/test/NeXusFileTest.h

```diff
@@ -465,6 +468,339 @@ class NeXusFileLeakTest : public CxxTest::TestSuite {
 
       // Delete file
       removeFile(szFile);
+      cout << "Leak Test 3 Success!\n";
+    }
+  }
+#endif
+
+  /**
+   * These tests correspond to tests inside napi_test.cpp
+   * Refactored to work as unit tests with asserts and comparisons
+   * as opposed to a single long print-out test
+   */
+
+private:
+  File do_prep_files(std::string const nxFile) {
+    removeFile(nxFile); // in case previous run didn't clean up
+
+    cout << "Creating \"" << nxFile << "\"" << endl;
+    // create file
+    File fileid(nxFile, NXACC_CREATE5);
+
+    fileid.makeGroup("entry", "NXentry");
+    fileid.openGroup("entry", "NXentry");
+    fileid.putAttr("hugo", "namenlos");
+    fileid.putAttr("cucumber", "passion");
+    return fileid;
+  }
+
+  template <typename T> void do_rw_test(File &fileid, std::string const &dataname, T const &data) {
+    cout << "Testing data " << dataname << "\n";
+    // write
+    fileid.writeData(dataname, data);
+
+    // read
+    T output;
+    fileid.readData(dataname, output);
+
+    // compare
+    TS_ASSERT_EQUALS(data, output);
+  }
+
+  template <size_t N, size_t M, typename T>
+  void do_rw2darray_test(File &fileid, std::string const &dataname, T const (&data)[N][M]) {
+    cout << "Testing attribute " << dataname << "\n";
+    // write
+    fileid.makeData(dataname, getType<T>(), DimVector({N, M}));
+    fileid.openData(dataname);
+    fileid.putData(data);
+    fileid.closeData();
+
+    // read
+    T output[N][M];
+    fileid.openData(dataname);
+    fileid.getData(&(output[0][0]));
+    fileid.closeData();
+
+    // compare
+    for (size_t i = 0; i < N; i++) {
+      for (size_t j = 0; j < M; j++) {
+        TS_ASSERT_EQUALS(data[i][j], output[i][j]);
+      }
+    }
+  }
+
+  template <typename T> void do_rwslabvec_test(File &fileid, std::string const dataname, vector<T> const &data) {
+    cout << "Testing slab " << dataname << "\n";
+
+    // write
+    dimsize_t dimsize = data.size();
+    DimSizeVector const start({0}), size({dimsize});
+    fileid.makeData(dataname, getType<T>(), dimsize);
+    fileid.openData(dataname);
+    fileid.putSlab(data, start, size);
+    fileid.closeData();
+
+    // read
+    T output[5];
```

Comment:
Please pull the magic number out into a constant to make the code easier to read